### PR TITLE
Git updates - haiku instructions

### PR DESCRIPTION
**Pull request by erikvansebille**
Adding instructions for the haikus in week 2

### CODE REVIEW

File: Lectures/Lecture2_GitforCollaboration.md
Comment (TerpstraS) [inline Lectures/Lecture2_GitforCollaboration.md:35]: Should the students create their own repository or their own branch on the course repository? In the assignment it mentions creating a branch of the course repository and here it mentions they should create their own repository.

**Review by TerpstraS — CHANGES_REQUESTED** [review]

File: Lectures/Lecture2_GitforCollaboration.md
Comment (erikvansebille) [reply Lectures/Lecture2_GitforCollaboration.md:35]: I meant a branch; good catch! ALthough we could also choose for the groups to fork the repository?

File: Resources/_template_week2_Haikus.md
Comment (TerpstraS) [inline Resources/_template_week2_Haikus.md:14]: Last year we allowed the students to create, review, and approve merge requests, but not to actually perform the merge. This is to prevent accidents that can mess up the whole course repository (because it allows us to quickly check the pull request before merging it). Maybe an idea to keep this the same as last year?

File: Lectures/Lecture2_GitforCollaboration.md
Comment (TerpstraS) [reply Lectures/Lecture2_GitforCollaboration.md]: Yes, that is also possible. Although that means they need to merge back the changes from their fork to the main repository. Maybe this will complicate things a bit too much?

Comment (erikvansebille) [reply Lectures/Lecture2_GitforCollaboration.md]: I don't think 'complication' is a criterium for not doing it. We want the students to go deep and challenge them?

File: Resources/_template_week2_Haikus.md
Comment (erikvansebille) [reply Resources/_template_week2_Haikus.md:14]: An idea could be to create three new repositories, for the Haikus of groups 1, 2 and 3; and let each subteam merge (and potentially break?) that repository. The main repository will then at least stay untouched. Is that an idea?

Comment (TerpstraS) [reply Resources/_template_week2_Haikus.md:14]: Yes, I like that idea! That should also be feasible.

File: Lectures/Lecture2_GitforCollaboration.md
Comment (TerpstraS) [reply Lectures/Lecture2_GitforCollaboration.md]: Yes, that's true. I still think that branching is the more important one to practice first, because we will also use that later on for other assignments. But yes, forking is also a possibility

File: Resources/_template_week2_Haikus.md
Comment (erikvansebille) [reply Resources/_template_week2_Haikus.md:14]: > Yes, I like that idea! That should also be feasible.

OK, what do the other think? @SanliFaez, do you agree that I create three repositories for the three groups?

File: Lectures/Lecture2_GitforCollaboration.md
Comment (SanliFaez) [inline Lectures/Lecture2_GitforCollaboration.md:32]: for people who have no experience with bash, some of the instructions in the notebook can be still quite confusing. For example, in lesson 1-setting up, I am not sure I understand myself the purpose of setting up the global username and the options in each command-line. 
The tutorial sometimes skips explaining why a command is necessary.

File: Lectures/Lecture2_GitforCollaboration.md
Comment (SanliFaez) [inline Lectures/Lecture2_GitforCollaboration.md:35]: can you be more explicit in the instructions? will the team of 3 work on the same branch but separate computers? does it mean all need to have their own clones? 
please specify "They" and "then" in the penultimate sentence.

File: Resources/_template_week2_Haikus.md
Comment (SanliFaez) [inline Resources/_template_week2_Haikus.md:8]: I think it helps if you use the git terminology: instead of upload, use pushes to the remote, and specify the "branch" is the subteams branch on remote.

File: Resources/_template_week2_Haikus.md
Comment (SanliFaez) [inline Resources/_template_week2_Haikus.md:14]: students have developer role, and not maintainer on gitlab. They cannot approve merge request on the main repo. This option is reserved for the maintainers: the instructors team

File: Resources/_template_week2_Haikus.md
Comment (SanliFaez) [inline Resources/_template_week2_Haikus.md:31]: I think it is better to work with git.science accounts, which are based on solisID. for most students this will be the same as student number, but not all.

**Review by SanliFaez — COMMENTED** [review]
I was not sure if my suggestions are good enough for directly implementing in the text. I think it is a very clear exercise but can use a bit more of git terminology

File: Resources/_template_week2_Haikus.md
Comment (SanliFaez) [reply Resources/_template_week2_Haikus.md:14]: I think that can be confusing. I restrict approval of merge requests to TAs, so all students will be developers and cannot merge to main.
therefore, I prefer to keep everything in just one repository, but restrict approving of merges to TAs.
if it is only for one exercise, you can also avoid a lot of conflicts by creating three sub-folders, one for each group. 

I think you still need to be more specific on where they need to add their subteam md files.

File: Resources/_template_week2_Haikus.md
Comment (SanliFaez) [reply Resources/_template_week2_Haikus.md:8]: step 3, specify they should copy to the /Assignments subfolder, and do not touch the /Resources.

File: Resources/_template_week2_Haikus.md
Comment (SanliFaez) [reply Resources/_template_week2_Haikus.md:14]: okay, I have created a subfolder for this exercise under assignments. If students each create their Haiku under a separate filename as instructure, the merge requests will have no conflict with each other, and the three TAs can each merge the requests from their group without causing a conflict.

**Comment by erikvansebille** [comment]
@TerpstraS, could you please rereview and approve, because your requested changes are now blocking merging

**Review by TerpstraS — APPROVED** [review]
Looks good! Just fixed a typo

**Pull request merged**
